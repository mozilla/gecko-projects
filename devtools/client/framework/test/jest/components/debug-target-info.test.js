/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

/**
 * Unit tests for the DebugTargetInfo component.
 */

const renderer = require("react-test-renderer");
const React = require("devtools/client/shared/vendor/react");
const DebugTargetInfo = React.createFactory(require("devtools/client/framework/components/DebugTargetInfo"));
const { CONNECTION_TYPES } = require("devtools/client/shared/remote-debugging/remote-client-manager");

/**
 * Stub for the L10N property expected by the DebugTargetInfo component.
 */
const stubL10N = {
  getStr: id => id,
  getFormatStr: (id, ...args) => [id, ...args].join("-"),
};

const findByClassName = (testInstance, className) => {
  return testInstance.findAll(node => {
    return node.props.className && node.props.className.includes(className);
  });
};

const TEST_TOOLBOX = {
  target: {
    name: "Test Tab Name",
    url: "http://some.target/url",
  },
};

const TEST_TOOLBOX_NO_NAME = {
  target: {
    url: "http://some.target/without/a/name",
  },
};

const USB_DEVICE_DESCRIPTION = {
  brandName: "usbRuntimeBrandName",
  connectionType: CONNECTION_TYPES.USB,
  channel: "release",
  deviceName: "usbDeviceName",
  version: "1.0.0",
};

const THIS_FIREFOX_DEVICE_DESCRIPTION = {
  brandName: "thisFirefoxRuntimeBrandName",
  connectionType: CONNECTION_TYPES.THIS_FIREFOX,
  channel: "release",
  version: "1.0.0",
};

const USB_TARGET_INFO = DebugTargetInfo({
  deviceDescription: USB_DEVICE_DESCRIPTION,
  toolbox: TEST_TOOLBOX,
  L10N: stubL10N,
});

const THIS_FIREFOX_TARGET_INFO = DebugTargetInfo({
  deviceDescription: THIS_FIREFOX_DEVICE_DESCRIPTION,
  toolbox: TEST_TOOLBOX,
  L10N: stubL10N,
});

const THIS_FIREFOX_NO_NAME_TARGET_INFO = DebugTargetInfo({
  deviceDescription: THIS_FIREFOX_DEVICE_DESCRIPTION,
  toolbox: TEST_TOOLBOX_NO_NAME,
  L10N: stubL10N,
});

describe("DebugTargetInfo component", () => {
  it("displays connection info for USB Release target", () => {
    const targetInfo = renderer.create(USB_TARGET_INFO);
    expect(findByClassName(targetInfo.root, "js-connection-info").length).toEqual(1);
  });

  it("renders the expected snapshot for USB Release target", () => {
    const targetInfo = renderer.create(USB_TARGET_INFO);
    expect(targetInfo.toJSON()).toMatchSnapshot();
  });

  it("hides the connection info for This Firefox target", () => {
    const targetInfo = renderer.create(THIS_FIREFOX_TARGET_INFO);
    expect(findByClassName(targetInfo.root, "js-connection-info").length).toEqual(0);
  });

  it("displays the target title if the target of the Toolbox has a name", () => {
    const targetInfo = renderer.create(THIS_FIREFOX_TARGET_INFO);
    expect(findByClassName(targetInfo.root, "js-target-title").length).toEqual(1);
  });

  it("renders the expected snapshot for This Firefox target", () => {
    const targetInfo = renderer.create(THIS_FIREFOX_TARGET_INFO);
    expect(targetInfo.toJSON()).toMatchSnapshot();
  });

  it("doesn't display the target title if the target of the Toolbox has no name", () => {
    const targetInfo = renderer.create(THIS_FIREFOX_NO_NAME_TARGET_INFO);
    expect(findByClassName(targetInfo.root, "js-target-title").length).toEqual(0);
  });

  it("renders the expected snapshot for a Toolbox with an unnamed target", () => {
    const targetInfo = renderer.create(THIS_FIREFOX_NO_NAME_TARGET_INFO);
    expect(targetInfo.toJSON()).toMatchSnapshot();
  });
});
