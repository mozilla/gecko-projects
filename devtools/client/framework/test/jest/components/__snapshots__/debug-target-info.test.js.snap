// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebugTargetInfo component renders the expected snapshot for This Firefox target 1`] = `
<header
  className="debug-target-info"
>
  <span
    className="iconized-label"
  >
    <img
      className="channel-icon"
      src="chrome://devtools/skin/images/aboutdebugging-firefox-release.svg"
    />
    <b
      className="devtools-ellipsis-text"
    >
      toolbox.debugTargetInfo.runtimeLabel.thisFirefox-1.0.0
    </b>
    <span
      className="devtools-ellipsis-text"
    />
  </span>
  <span
    className="iconized-label"
  >
    <img
      alt="favicon"
      src="chrome://devtools/skin/images/aboutdebugging-globe-icon.svg"
    />
    <b
      className="devtools-ellipsis-text js-target-title"
    >
      Test Tab Name
    </b>
    <span
      className="devtools-ellipsis-text"
    >
      http://some.target/url
    </span>
  </span>
</header>
`;

exports[`DebugTargetInfo component renders the expected snapshot for USB Release target 1`] = `
<header
  className="debug-target-info"
>
  <span
    className="iconized-label js-connection-info"
  >
    <img
      alt="usb icon"
      src="chrome://devtools/skin/images/aboutdebugging-usb-icon.svg"
    />
    toolbox.debugTargetInfo.connection.usb
  </span>
  <span
    className="iconized-label"
  >
    <img
      className="channel-icon"
      src="chrome://devtools/skin/images/aboutdebugging-firefox-release.svg"
    />
    <b
      className="devtools-ellipsis-text"
    >
      toolbox.debugTargetInfo.runtimeLabel-usbRuntimeBrandName-1.0.0
    </b>
    <span
      className="devtools-ellipsis-text"
    >
      usbDeviceName
    </span>
  </span>
  <span
    className="iconized-label"
  >
    <img
      alt="favicon"
      src="chrome://devtools/skin/images/aboutdebugging-globe-icon.svg"
    />
    <b
      className="devtools-ellipsis-text js-target-title"
    >
      Test Tab Name
    </b>
    <span
      className="devtools-ellipsis-text"
    >
      http://some.target/url
    </span>
  </span>
</header>
`;

exports[`DebugTargetInfo component renders the expected snapshot for a Toolbox with an unnamed target 1`] = `
<header
  className="debug-target-info"
>
  <span
    className="iconized-label"
  >
    <img
      className="channel-icon"
      src="chrome://devtools/skin/images/aboutdebugging-firefox-release.svg"
    />
    <b
      className="devtools-ellipsis-text"
    >
      toolbox.debugTargetInfo.runtimeLabel.thisFirefox-1.0.0
    </b>
    <span
      className="devtools-ellipsis-text"
    />
  </span>
  <span
    className="iconized-label"
  >
    <img
      alt="favicon"
      src="chrome://devtools/skin/images/aboutdebugging-globe-icon.svg"
    />
    <span
      className="devtools-ellipsis-text"
    >
      http://some.target/without/a/name
    </span>
  </span>
</header>
`;
